### ISSUE TITLE
Require SPDX

### ISSUE THREAD

**Issue by isuruf** [conda-forge/conda-smithy#1299]
This has been working well for a while now. This would be a linter failure instead of a hint.

cc @conda-forge/core

**Comment by CJ-Wright** [comment]
What is the plan to bring failing feedstocks into compliance?

**Comment by isuruf** [comment]
Nothing. Maintainers should fix them.

**Comment by beckermr** [comment]
We can propose migrations for a significant fraction of them.

**Comment by isuruf** [comment]
I'd caution against automating it because they can be wrong especially with GPL-3.0-or-later vs GPL-3.0-only.

**Comment by beckermr** [comment]
Then what do we do about the R migration? Should we ignore the linter for all of that?

**Comment by isuruf** [comment]
For R, the information is given to us by R. We can use that.

**Comment by h-vetinari** [comment]
What about those that cannot be made compatible with the current rule? Especially those that combine several licenses, or worse, have different licenses for different parts of the codebase.

Recent example I've worked on [cryptography](https://github.com/conda-forge/cryptography-feedstock/pull/37#issuecomment-618993610) (neither `AND` nor `OR`-conjunctions between SPDX identifiers worked), [suitesparse](https://github.com/conda-forge/conda-smithy/pull/1220#issuecomment-612638751) (more complicated expression).

**Comment by isuruf** [comment]
cryptography works fine as far as I can see.

**Comment by isuruf** [comment]
Can you open an issue for suitesparse?

**Comment by h-vetinari** [comment]
> @isuruf: cryptography works fine as far as I can see.

I don't think it does. The linter checks every commit but doesn't reissue the warning if the warning hasn't changed (or been cleared). If you look at https://github.com/conda-forge/cryptography-feedstock/pull/37 you'll see that the warning was never cleared. It would be also easy to see by just opening a new PR.

**Comment by h-vetinari** [comment]
> @isuruf: Can you open an issue for suitesparse?

#1300

**Comment by jakirkham** [comment]
This is done, right? Should we close?

**Comment by beckermr** [comment]
Where is the PR?

**Comment by djhoese** [comment]
Found this while working on the HDF5 package which has its own license (apparently) so "HDF5" is listed: https://github.com/conda-forge/hdf5-feedstock

**Comment by isuruf** [comment]
@djhoese, what's the issue here?

**Comment by djhoese** [comment]
That the HDF5 package wouldn't pass the linter with its current license, right? Maybe I misunderstood this issue, but it sounded like the linter would require SPDX license identifiers including compound/combined ones. The HDF5 library doesn't fit in to that.

**Comment by isuruf** [comment]
Linter posts a comment like https://github.com/conda-forge/hdf5-feedstock/pull/122#issuecomment-660549438 which has a link to docs https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions . Docs specifically mentions HDF5.

**Comment by isuruf** [comment]
Btw, PRs are welcome to improve docs.

**Comment by djhoese** [comment]
> PRs are welcome to improve docs.

Mostly just inexperience with SPDX identifiers and this was the first time running into a non-SPDX license on a conda-forge package (I'm not a maintainer of HDF5). I probably just rushed to comment and should have read more. Thanks for the info.

**Comment by isuruf** [comment]
That's totally fine. Docs become better when others bring a fresh perspective.

**Comment by jakirkham** [comment]
> Where is the PR?

Sorry Matt. Looks like I never answered this.

It was done in PR: https://github.com/conda-forge/conda-smithy/pull/1220

Though there are some follow on PRs linked as well.

**Comment by isuruf** [comment]
Hmm, it looks like it's not required yet
